### ISSUE TITLE
Android intent cannot parse content:// scheme

### ISSUE THREAD

**Issue by rossbeto** [mwrlabs/drozer#271]
### Steps to reproduce
1.intall drozer 2.4.2
2. run app.activity.start --action android.intent.action.VIEW  --component [XXX] [YYY] --data-uri content://gmail-ls/XXXMail@gmail.com/messages/2/attachments/0.1/BEST/false --mimetype application/txt
3. Inspect the intent that is being passed to android application
4. intent.getData() return null

### Expected behaviour
 intent.getData() == "content://gmail-ls/XXXMail@gmail.com/messages/2/attachments/0.1/BEST/false"

in Android app, if we do Uri.parse("content://gmail-ls/XXXMail@gmail.com/messages/2/attachments/0.1/BEST/false"), it returns the correct Uri.

### Actual behaviour
 intent.getData() == null

### Environment

**Operating system**:
OS X EL Capital (10.11.6 (15G1004))
**Java Version:**
java version "1.8.0_66"
**Android Version:**
API 24 
**Drozer Version:** 
2.4.2
**Install Method:** Built from source/.deb/.exe/..
install from .egg package

**Comment by HenryHoggard** [comment]
Hi @rossbeto  have you tried wrapping the data-uri in quotes?

**Comment by rossbeto** [comment]
@HenryHoggard 👍  thanks for the reply. I found out that it was my mistake. By default " run app.activity.start" create an empty Extras bundle for the intent ( intent.getExtras().size == 0 , it is not null ), and there is a check in my application that based on this Extras bundle is null...
So there is nothing wrong with the Intent in this case, I passed the content as it is ( no quotes needed). This issue should be closed.
However, I still did not manage to consume the Intent in this way. I just want to share the situation : 
+ When I tapped on the attachment in mail, and share it with my application then it works fine.
+ If I do it from Drozer like this:
`run app.activity.start --action android.intent.action.VIEW --data-uri  content://gmail-ls/XXXMail@gmail.com/messages/2/attachments/0.1/BEST/false --component android com.android.internal.app.ResolverActivity  --flags ACTIVITY_NEW_TASK`
Then I choose my app to consume this Intent. I run into a Security exception (because of lacking runtime permission )

`Unable to start activity ComponentInfo{PACKAGENAME/ ActivityName}: java.lang.SecurityException: Permission Denial: opening provider com.google.android.gm.provider.MailProvider from ProcessRecord{5f99a60 5314:PACKAGENAME/u0a104} (pid=5314, uid=10104) requires com.google.android.gm.permission.READ_GMAIL or com.google.android.gm.permission.WRITE_GMAIL
`
I guess that I will need to find a way to grant permission for this pid.
cheer.
